examples/bri-3/src/bri/merkleTree/agents/merkleTreeStorage.agent.ts

```diff
@@ -0,0 +1,72 @@
+import { Injectable } from '@nestjs/common';
+import { InjectMapper } from '@automapper/nestjs';
+import { Mapper } from '@automapper/core';
+import { PrismaService } from '../../../../prisma/prisma.service';
+import MerkleTree from 'merkletreejs';
+import { BpiMerkleTree } from '../models/bpiMerkleTree';
+
+@Injectable()
+export class MerkleTreeStorageAgent extends PrismaService {
+  constructor(@InjectMapper() private mapper: Mapper) {
+    super();
+  }
+
+  async getMerkleTreeById(id: string): Promise<BpiMerkleTree> {
+    const merkleTreeModel = await this.bpiMerkleTree.findUnique({
+      where: { id },
+    });
+
+    if (!merkleTreeModel) {
+      return null;
+    }
+
+    const merkleTree = new BpiMerkleTree(
+      merkleTreeModel.id,
+      MerkleTree.unmarshalTree(merkleTreeModel.merkleTree),
+    );
+    return this.mapper.map(merkleTree, BpiMerkleTree, BpiMerkleTree);
+  }
+
+  async createNewMerkleTree(merkleTree: BpiMerkleTree): Promise<BpiMerkleTree> {
+    const treeIn = {
+      id: merkleTree.id,
+      merkleTree: MerkleTree.marshalTree(merkleTree.merkleTree),
```

Comment:
the naming convention : merkleTree.merkleTree can be improved to merkleTree.tree.

Reply:
sounds like a good suggestion. Unless some other naming scheme comes to mind will likely go with that.